src/components/PostDetails.tsx

```diff
@@ -82,19 +64,17 @@
     // we delete the comment immediately so as
     // not to make the user wait long for the actual deletion
     // eslint-disable-next-line max-len
-    setComments(currentComments =>
-      currentComments.filter(comment => comment.id !== commentId),
-    );
+    setComments(comments.filter(comment => comment.id !== commentId));
```

Comment:
Similarly, the `setComments` function here should be dispatched as an action. Use `dispatch(setComments(comments.filter(comment => comment.id !== commentId)));` to update the Redux state correctly.